python/interpret_community/widget/explanation_dashboard.py

```diff
@@ -0,0 +1,183 @@
+                sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+                sock.bind(("::1", port))
+                sock.listen(backlog)
+                sock.close()
+            except socket.error:  # pragma: no cover
+                if rais:
+                    error_message = """Port {0} is not available.
+                    Please specify another port for use via the 'port' parameter"""
+                    raise RuntimeError(
+                        error_message.format(port)
+                    )
+                else:
+                    return False
+            return True
+
+        @app.route('/')
+        def hello():
+            return "No global list view supported at this time."
+
+        @app.route('/<id>')
+        def explanation_visual(id):
+            # if there is no internet or the static file exists, use that
+            # main_js='http://127.0.0.1:5000/static/index.js'
+            if id in ExplanationDashboard.explanations:
+                using_fallback = False
+                if ExplanationDashboard.service.use_cdn:
+                    try:
+                        url = 'https://interpret-cdn.azureedge.net/{0}'.format(ExplanationDashboard._cdn_path)
+                        r = requests.get(url)
+                        if not r.ok:
+                            using_fallback = True
+                            url = "http://{0}:{1}/static/index.js".format(
+                                ExplanationDashboard.service.ip,
+                                ExplanationDashboard.service.port)
+                    except Exception:
+                        using_fallback = True
+                        url = "http://{0}:{1}/static/index.js".format(
+                            ExplanationDashboard.service.ip,
+                            ExplanationDashboard.service.port)
+                else:
+                    url = "http://{0}:{1}/static/index.js".format(
+                        ExplanationDashboard.service.ip,
+                        ExplanationDashboard.service.port)
+                serialized_explanation = json.dumps(ExplanationDashboard.explanations[id].dashboard_input)
+                return render_template('dashboard.html',
+                                       explanation=serialized_explanation,
+                                       main_js=url,
+                                       app_id='app_123',
+                                       using_fallback=using_fallback)
+            else:
+                return "Unknown model id."
+
+        @app.route('/<id>/predict', methods=['POST'])
+        def predict(id):
+            data = request.get_json(force=True)
+            if id in ExplanationDashboard.explanations:
+                return ExplanationDashboard.explanations[id].on_predict(data)
+
+    def __init__(self, explanation, model=None, *, dataset=None,
+                 true_y=None, classes=None, features=None, port=5000, use_cdn=True):
```

Comment:
for the port, you could just try and get a free one instead of setting it to 5000.  also, what happens if multiple dashboards run at once?

Reply:
Multiple dashboards are supported as different routes served out of the same port, right now ID'd by the index number in when they were displayed. I have a task for finding an available port rather than failing with a message, but thought it wasn't necessary for the initial PR. Can add it if it seems necessary?